### PR TITLE
Fix: Remap e-component instance IDs on website template import [ED-25468]

### PR DESCRIPTION
**Pull request by IshayMaya**
## PR Checklist
- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Fix: Website template import now remaps component instance IDs to the destination site [ED-25468]

## Description

Kit import saved `e-component` widgets with the **source-site** `component_id`. `Document::on_import_update_dynamic_content()` only remaps legacy `__dynamic__` settings, so atomic `component_instance` props were left pointing at posts that do not exist on the destination.

That breaks any imported document that instances a component (pages, Theme Builder archives, Collection Loop items). Loop items looked empty in Structure; Enter Component Edit failed on pages.

`Import::save_elements_of_imported_posts()` now remaps `settings.component_instance.value.component_id.value` using the existing source→destination `post_ids` map before save.

Related follow-up (not in this PR): [ED-25469](https://elementor.atlassian.net/browse/ED-25469) — loop query validation can still fatal on import (`Settings validation failed. query: invalid_value`).

## Test instructions
This PR can be tested by following these steps:

1. On a source site, create a page and/or Archive template with an `e-component` instance (including inside a Collection Loop item). Export **Website Template** with Content, Templates, and Settings.
2. On a destination site (fresh, or after **Remove Website Template**), import the ZIP.
3. Open the imported Archive: Loop Card should still show the Component in Structure.
4. Open the imported page: Enter Component Edit should work on the instance.

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [x] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes [ED-25468](https://elementor.atlassian.net/browse/ED-25468)

Made with [Cursor](https://cursor.com)

[ED-25468]: https://elementor.atlassian.net/browse/ED-25468?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ED-25469]: https://elementor.atlassian.net/browse/ED-25469?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## 1. Problem & Context

Website template imports serialize e-component widget IDs pointing to source-site posts, but need remapping to destination-site post IDs post-import. This ensures imported components reference correct posts after creation.

## 2. What Changed (Where)

- Added `Remap_Component_Instance_Ids` utility class traversing element trees, identifying e-component widgets, and updating `component_id` via post ID mapping.
- Integrated remapping into both import runners (customization + legacy) before document save.
- Added `set_component_id()` helper to `Component_Instance_Prop_Type` for settings mutation.
- Comprehensive unit tests covering nested/unmapped components and edge cases.

## 3. How It Works

Import runner calls `Remap_Component_Instance_Ids::apply(elements, post_ids_map)` → recursively traverses element tree → identifies e-component widgets via `elType='widget'` + `widgetType='e-component'` → extracts original component ID → looks up mapping → updates via `set_component_id()` → returns mutated tree before document save. Empty map short-circuits immediately.

## 4. Risks

None identified. Remapping is idempotent and only affects component instance references. Non-matching IDs pass through unchanged. Test coverage validates nested structures and ignores unrelated widgets.

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using.
💡 **Tip:** You can customize your AI Description using **Guidelines** [Learn how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>

### CODE REVIEW

File: app/modules/import-export-customization/processes/import.php [hunk 1]
```diff
@@ -784,6 +784,7 @@ private function save_elements_of_imported_posts() {
 			$document = Plugin::$instance->documents->get( $new_id );
 
 			if ( isset( $data['elements'] ) ) {
```
Comment (Omerisra6) [inline app/modules/import-export-customization/processes/import.php:787]: maybe we can iterate once and do both mutations?

File: app/modules/import-export-customization/processes/import.php [hunk 2]
```diff
@@ -801,6 +802,45 @@ private function save_elements_of_imported_posts() {
...
+	 * Rewrite `component_id` inside every `e-component` widget so it points at the
+	 * component post that was recreated on the destination site instead of the
+	 * source-site post id that was serialized in the exported template.
+	 *
+	 * @param array $elements Elements tree from the exported document.
+	 * @param array $post_ids_map Source→destination post id map produced by the runner.
+	 *
+	 * @return array
+	 */
+	private function remap_component_instances( array $elements, array $post_ids_map ): array {
```
Comment (Omerisra6) [inline app/modules/import-export-customization/processes/import.php]: maybe this should come from the components module?

File: app/modules/import-export-customization/processes/import.php [hunk 3]
```diff
@@ -801,6 +802,45 @@ private function save_elements_of_imported_posts() {
...
+			$is_component_widget = 'widget' === ( $element['elType'] ?? null )
+				&& 'e-component' === ( $element['widgetType'] ?? null );
+
+			if ( $is_component_widget ) {
+				$source_id = $element['settings']['component_instance']['value']['component_id']['value'] ?? null;
+
+				if ( is_numeric( $source_id ) && isset( $post_ids_map[ (int) $source_id ] ) ) {
+					$element['settings']['component_instance']['value']['component_id']['value'] = (int) $post_ids_map[ (int) $source_id ];
+				}
+			}
```
Comment (Omerisra6) [inline app/modules/import-export-customization/processes/import.php]: I think we have utils/consts for these in the module

**Review by Omerisra6 — APPROVED** [review]
